app/src/main/kotlin/com/hippo/ehviewer/ui/main/AvatarIcon.kt

```diff
@@ -0,0 +1,201 @@
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.LinearProgressIndicator
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import arrow.core.Either
+import arrow.core.Either.Companion.catch
+import arrow.core.Option
+import arrow.core.none
+import arrow.core.some
+import coil3.compose.AsyncImage
+import com.hippo.ehviewer.R
+import com.hippo.ehviewer.Settings
+import com.hippo.ehviewer.client.EhEngine
+import com.hippo.ehviewer.client.EhUtils
+import com.hippo.ehviewer.client.parser.HomeParser
+import com.hippo.ehviewer.collectAsState
+import com.hippo.ehviewer.ui.tools.DialogState
+import com.hippo.ehviewer.util.displayString
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.conflate
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.merge
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import moe.tarsin.coroutines.runSwallowingWithUI
+
+private val limitScope = CoroutineScope(Dispatchers.IO)
+private val refreshEvent = MutableSharedFlow<Unit>()
+private val invalidateEvent = MutableSharedFlow<Unit>()
+
+typealias Result = Option<Either<String, HomeParser.Result>>
+
+private val limitFlow: StateFlow<Result> = refreshEvent.conflate()
+    .onEach {
+        // If cached value is error, drop it when refresh
+        val isError = limitFlow.value.isSome { it.isLeft() }
+        if (isError) invalidateEvent.emit(Unit)
+    }
+    .map { catch { EhEngine.getImageLimits() }.mapLeft { e -> e.displayString() } }
+    .let { src -> merge(src.map { v -> v.some() }, invalidateEvent.map { none() }) }
+    .stateIn(limitScope, SharingStarted.Eagerly, none())
+    .apply {
+        limitScope.launch {
+            collectLatest {
+                // Consider cached value needs invalidate for 2 minutes
+                delay(120.seconds)
+                invalidateEvent.emit(Unit)
```

Comment:
It works but feels clunky.

Reply:
Maybe we just not invalidate cache, and bonus animated quota